File: Practica_01/BubbleSort/BubbleSort.py
from time import time

def lectura(a,k):
    archivo = open("../valores_10_5.txt","r")
    avance = 0
    for linea in archivo.readlines():
        if avance == k:
            break
        array = linea
        a.append(int(array))
        avance += 1
    archivo.close()


def bubbleSort(a):
    n = len(a)
    for i in range(n):
        swapped = False
        for j in range(0, n-i-1):
            if a[j] > a[j+1] :
                temp = a[j]
                a[j] = a[j+1]
                a[j+1] = temp
                swapped = True
        if swapped == False:
            break

def set_data(tamanho,tiempo):

  helper = open("valores_python.txt","a");
  helper.write(str(tamanho) + " , "+ str(tiempo) + "\n")
  helper.close();


a = []
intervalo = 2000
for k in range(2000,100001,intervalo):
    lectura(a,k)
    start_time = time()
    bubbleSort(a)
    elapsed_time = time() - start_time
    set_data(len(a),round(float(elapsed_time),4))
    #print("Elapsed time: %0.10f seconds." % elapsed_time)
    del a[:]
    intervalo *= 2

File: Practica_01/BubbleSort/BubbleSort.cpp
#include <iostream>
#include <vector>
#include <ctime>
#include <fstream>
#include <iomanip>
using namespace std;

void swap(int &a, int &b)
{
    int temp = a;
    a = b;
    b = temp;
}

void bubbleSort(int a[],int b)
{
	  int tamanho = b;
  	bool swapped;
    for (int i = 0; i < tamanho-1; i++)
    {
    	swapped = false;
    	for (int j = 0; j < tamanho-i-1; j++)
    	{
    		if (a[j] > a[j+1])
    		{
    			swap(a[j], a[j+1]);
    			swapped = true;
    		}
    	}
    	if(!swapped) break;
    }
}

void agregar_valor(int a[],int carga)
{
    fstream texto("../valores_10_5.txt");
    int aux;
    for (int i = 0; i < carga; i++)
    {
        texto >> aux;
        a[i] = aux;
    }
    texto.close();
}
/*
void bubbleSort(vector<int>&a)
{
	  int tamanho = a.size();
  	bool swapped;
    for (int i = 0; i < tamanho-1; i++)
    {
    	swapped = false;
    	for (int j = 0; j < tamanho-i-1; j++)
    	{
    		if (a[j] > a[j+1])
    		{
    			swap(a[j], a[j+1]);
    			swapped = true;
    		}
    	}
    	if(!swapped) break;
    }
}


void agregar_valor(vector<int> &a,int carga)
{
    fstream texto("../valores_10_5.txt");
    int aux;
    for (int i = 0; i < carga; i++)
    {
        texto >> aux;
        a.push_back(aux);
    }
    texto.close();
}
*/

void set_data(int tamanho,float tiempo)
{
  fstream helper;
  helper.open("valores_C++_2.txt");
  helper.seekg (0,helper.end);
  helper << tamanho << " , ";
  helper<<fixed<<setprecision(5)<<tiempo<<endl;
  helper.close();
}


int main()
{
	//vector<int> valores;
  int valores[100000];
  int val = 10;
  float t0,t1;
  for(int i= 10; i < 100000;i += val)
  {
    agregar_valor(valores,i);
    t0 = clock();
    bubbleSort(valores,i);
    t1 = clock();
    float tiempo = ((t1-t0)/CLOCKS_PER_SEC);
    set_data(i,tiempo);
    //valores.clear();
    val *= 2;
  }
  //valores.clear();
  agregar_valor(valores,100000);
  t0 = clock();
  bubbleSort(valores,100000);
  t1 = clock();
  float tiempo = ((t1-t0)/CLOCKS_PER_SEC);
  set_data(100000,tiempo);


}
